### ISSUE TITLE
Allow route handlers to be overridden

### ISSUE THREAD

**Issue by jessekelly881** [sukovanej/effect-http#308]
It might be useful to allow already handled routes to be overridden. Especially when you are already using ExampleServer for testing and want to progressively add real implementations. 

I came across this when I was trying to use the Example server and also return `Metric.snapshot`.  E.g.

```ts
pipe(
  ExampleServer.make(Api.api),
  RouterBuilder.handle("metrics", () => Metric.snapshot) // error
)
```

**Comment by sukovanej** [comment]
Interesting idea 🤔. So far, I've been using the example server solely to check the swagger UI. Right now, I'm not sure we'd want to allow generally to override a handler in the composition chain. We'd loose the type-safety ensuring the server impl satisfies the `Api` contract.

**Comment by sukovanej** [comment]
Hey, I gave it a thought and it might be better to introduce the possibility of partially implementing the server on the `ExampleServer` side. I might expose a method `ExampleServer.handle`

```ts
pipe(
  RouterBuilder.make(api),
  ExampleServer.handle('myMethod1'),
  ExampleServer.handle('myMethod2'),
  RouterBuilder.handle("metrics", () => Metric.snapshot)
)
```

or perhaps `pipe(api, ExampleServer.makePartial('myMethod1', 'myMethod2', ...))`.

That way, I wouldn't need to introduce additional type mechanics (there's already enough of it 😄) and give the user-land code to partially implement some handlers using an example impl and some using the actual impl. WDYT about that?

Also, I think I'll rename the `ExampleServer` to something like `ExampleRouterBuilder` or something similar capturing the module more precisely.

**Comment by sukovanej** [comment]
Another option:

```ts
pipe(
  RouterBuilder.make(api),
  RouterBuilder.handle("metrics", () => Metric.snapshot),
  ExampleServer.handleRemaining,
)
```

**Comment by sukovanej** [comment]
Released in [0.45.1](https://github.com/sukovanej/effect-http/releases/tag/v0.45.1)
